src/Sepes.Infrastructure/Service/StudyParticipant/StudyParticipantLookupService.cs

```diff
@@ -72,36 +74,31 @@ public async Task<IEnumerable<ParticipantLookupDto>> GetLookupAsync(string searc
 
             var usersFromDbTask = _db.Users.Where(u => u.EmailAddress.StartsWith(searchText) || u.FullName.StartsWith(searchText) || u.ObjectId.Equals(searchText)).ToListAsync(cancellationToken);
 
+           return await MergeSearchResults(usersFromDbTask, usersFromAzureAdTask);
+           
+        }
+
+        async Task<IEnumerable<ParticipantLookupDto>> MergeSearchResults(Task<List<User>> usersFromDbTask, Task<Dictionary<string, AzureUserDto>> usersFromAzureAdTask)
+        {
             await Task.WhenAll(usersFromDbTask, usersFromAzureAdTask);
 
             var usersFromDb = _mapper.Map<IEnumerable<ParticipantLookupDto>>(usersFromDbTask.Result);
-            var usersFromDbAsDictionary = new Dictionary<string, ParticipantLookupDto>();
-
-            foreach (var curUserFromDb in usersFromDb)
-            {
-                if (string.IsNullOrWhiteSpace(curUserFromDb.ObjectId))
-                {
-                    continue;
-                }
-
-                if (!usersFromDbAsDictionary.ContainsKey(curUserFromDb.ObjectId))
-                {
-                    usersFromDbAsDictionary.Add(curUserFromDb.ObjectId, curUserFromDb);
-                }
-            }
+            var usersFromDbAsDictionary = usersFromDb.ToDictionary(u => u.ObjectId, u => u); 
 
             if (usersFromAzureAdTask.IsCompletedSuccessfully)
             {
-                var usersFromAzureAd = _mapper.Map<IEnumerable<ParticipantLookupDto>>(usersFromAzureAdTask.Result).ToList();
-
-                foreach (var curAzureUser in usersFromAzureAd)
+                foreach (var curAzureUser in usersFromAzureAdTask.Result)
                 {
-                    if (!usersFromDbAsDictionary.ContainsKey(curAzureUser.ObjectId))
+                    if (!usersFromDbAsDictionary.ContainsKey(curAzureUser.Key))
```

Comment:
Is ObjectID == key?

Reply:
True, since it's a dictionary, where objectid is the key